### PR TITLE
nixos: handle Systemd unit reload and restart triggers

### PR DESCRIPTION
**Pull request by NotAShelf**
This is a port of my personal reload/restart trigger system, detached from the manifest parser that I ended up writing. *Mostly* streamlined, but the logic might not be as robust as we desire. Partially addresses https://github.com/feel-co/hjem/issues/63

tl;dr: Introduces a new mechanism for automatically reloading or restarting systemd user units when their configuration changes. I've added test cases to make sure they actually work right, but there might be edge cases that I'm not actually seeing here.

### CODE REVIEW

**Review by GetPsyched — APPROVED** [review]
Not sure about the tests, but the systemd service looks decent.

Thanks for working on this :)

File: modules/nixos/base.nix [hunk 1]
```diff
@@ -201,11 +202,90 @@ in {
...
+          "hjem-reload@" = {
+            description = "Reload systemd user units for %i after Hjem file activation";
+            enableStrictShellChecks = true;
+            serviceConfig = {
+              User = "%i";
+              Type = "oneshot";
+            };
+            requires = ["hjem-activate@%i.service"];
+            after = ["hjem-activate@%i.service"];
+            path = [pkgs.jq pkgs.gnugrep];
```
Comment (eclairevoyant) [inline modules/nixos/base.nix]:
```suggestion
            path = [pkgs.coreutils-full pkgs.jq pkgs.gnugrep];
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: modules/nixos/base.nix [hunk 2]
```diff
@@ -201,11 +202,90 @@ in {
...
+            scriptArgs = "%i";
+            script = ''
+              ${checkEnabledUsers}
+
+              # XXX: This assumes that the XDG runtime directory is /run/user/<uid> which is correct
+              # *most of the time* but we cannot guarantee it. In the future we should try to infer
+              # and respect the existing runtime directory.
+              uid=$(id -u)
+              XDG_RUNTIME_DIR="/run/user/$uid"
+              export XDG_RUNTIME_DIR
```
[the remark this answers is not in the document]
Comment (NotAShelf) [reply modules/nixos/base.nix]: Same as above. The assign-then-export pattern is "required" by shellcheck.

**Review by eclairevoyant — APPROVED** [review]

**Comment by NotAShelf** [comment]
Merge on green...
